### PR TITLE
Replace reproducer for issue 1300 with minimal data file

### PR DESCRIPTION
**Pull request by dennisschagt**
Resolves https://github.com/newsboat/newsboat/issues/2748

I verified that the test reproduces the error on r2.21: availabe on [branch minimal-regression-test-on-r2.21 in my fork](https://github.com/dennisschagt/newsboat/tree/minimal-regression-test-on-r2.21), specifically on [commit 080bf1ef76](https://github.com/dennisschagt/newsboat/commit/080bf1ef76cfd93ba5f7a10f900e4880afc04785).
The 2 commits on top of it, on the linked branch, cherry-pick @Minoru's original fix and make the regression test pass.
The few commits below it were necessary to get r2.21 to compile with the newer compilers on my system.

### CODE REVIEW

File: test/htmlrenderer.cpp [hunk 1]
```diff
@@ -1621,14 +1621,15 @@ TEST_CASE("Skips contents of <script> tags", "[HtmlRenderer]")
 	std::vector<std::pair<LineType, std::string>> lines;
 	Links links;
 
-	std::ifstream input_file("data/1300-reproducer.html");
-	std::string input;
-	input_file >> input;
```
Comment (dennisschagt) [inline test/htmlrenderer.cpp:1626]: If I use this operator, the test fails.
I didn't look into it further, but I expect this only reads in a single line from the input file.
From a quick check, it looks like this was the only occurrence in our test suite.

I've added the `REQUIRE(lines[0].second == "Visible text");` line to make sure we actually used the full file.

Comment (Minoru) [reply test/htmlrenderer.cpp:1626]: It only read the first word, i.e. `<script>`. I don't understand why a single tag could trip up the renderer; perhaps I reproduced the bug manually, fixed it, then wrote a test without checking if it fails without the fix?

Anyway, I see you checked the updated code against 2.21 and it fails there, so everything's well.

**Review by Minoru — APPROVED** [review]
Thank you!

File: test/htmlrenderer.cpp (hunk 1 above)
Comment (dennisschagt) [reply test/htmlrenderer.cpp:1626]: > perhaps I reproduced the bug manually, fixed it, then wrote a test without checking if it fails without the fix?

No, the test you originally checked in #1302 does reproduce the error.
I used the original version of the test as the basis for the test in my first commit, but had to change it due to [a compilation warning](https://github.com/newsboat/newsboat/runs/24971253710).
It looks like the incorrect line was [introduced to fix that same kind of compilation issue](https://github.com/newsboat/newsboat/commit/6a759bc63db475c59fec2693c4658555f6e6b884#diff-e9f703a8e30d5a4a870c46322a6e2b499e8384a31db74f0557eb6fb63de949c9).

**Pull request merged**